tests/e2e/unified_test.go

```diff
@@ -186,7 +163,58 @@ func TestAllInSeries(t *testing.T) {
 					h.CompareGoldenFile(expectedPath, string(output), h.IgnoreComments, h.ReplaceString(project.ProjectID, "example-project-id"))
 				}
 
-				create.DeleteResources(h, s.Resources)
+				create.DeleteResources(h, opt.Create)
+
+				// Verify events against golden file
+				if os.Getenv("GOLDEN_REQUEST_CHECKS") != "" {
```

Comment:
nit: just a thought no need to make changes rn. I feel instead of saying `env != ""`, it's better to say `env == true`.

Reply:
The gotcha is that it's a string, but I agree it could be confusing, I'm just not sure what we should accept.  Maybe "0" and "1", and then being strict and failing the test if the value is specified but not 0 or 1?  We could wrap that logic into a helper function and evolve it.

This is "only test code", it probably is more important to think through the ergonomics if we support "feature flags" in the production codepaths.  But yes, I don't think this is perfect, I just don't know what's better!